### PR TITLE
Fix #484: Final document verification

### LINKED ISSUE #484: Final document verification
Single verification ID common to all documents should be send to Onboarding Provider _client evaluation_. `uploadSessionId` is most probably useless.

logs

```
DocumentProcessingService  : Verify document a92c911d-0c7f-41b7-b504-2998e449f4b2, uploaded-9b8630a2-a827-4c27-905b-...
DocumentProcessingService  : Verify document b8b1bec1-1380-4125-b5c3-bfb2fcba9f61, uploaded-4986d63b-6cd7-4dcd-afcf-...
IdentityVerificationService  : Verify documents 1a5ea207-1608-4d83-aec7-c506317253b0, [uploaded-9b8630a2-a827-4c27-905b-..., uploaded-4986d63b-6cd7-4dcd-afcf-...]
DocumentProcessingService  : Verify document 1cc2fb6a-a4c3-4db8-9aeb-6686681a87f2, uploaded-2cd8c7f9-0e9f-4e38-87c2-...
IdentityVerificationService  : Verify documents 97ae0e17-3260-406a-8e42-5b5e18b5f7d7, [uploaded-2cd8c7f9-0e9f-4e38-87c2-...]
MockOnboardingProvider      : Evaluating client for process ID: d9d4ff30-a5ee-4b00-b2ee-df6d726a62f1, user ID: mockuser_425986498155068775, verification ID: 1a5ea207-1608-4d83-aec7-c506317253b0, identityVerification ID: b056af51-fc49-4c76-b536-a041d5766d34
```

`es_document_verification`

```csv
id,activation_id,identity_verification_id,type,side,other_side_id,provider_name,status,filename,upload_id,verification_id,photo_id,verification_score,reject_reason,error_detail,original_document_id,used_for_verification,timestamp_created,timestamp_uploaded,timestamp_verified,timestamp_disposed,timestamp_last_updated,error_origin,reject_origin
47fe9ce0-2fef-4314-a6a3-c1d4d13058e2,57ebeba5-d94c-425d-9e3d-b4e1f675119e,b056af51-fc49-4c76-b536-a041d5766d34,DRIVING_LICENSE,FRONT,,mock,ACCEPTED,driving_license_mock_front.png,uploaded-2cd8c7f9-0e9f-4e38-87c2-...,97ae0e17-3260-406a-8e42-5b5e18b5f7d7,extracted-photo-id,,,,,true,2022-11-01 09:36:36.985000,2022-11-01 09:36:36.985000,2022-11-01 09:36:50.028000,,2022-11-01 09:36:50.028000,,
0c9759d2-3c4f-45df-b283-504b62d27d3b,57ebeba5-d94c-425d-9e3d-b4e1f675119e,b056af51-fc49-4c76-b536-a041d5766d34,SELFIE_PHOTO,,,mock,ACCEPTED,selfie_photo.jpg,uploaded-selfie-photo-57ebeba5-d9...,,,,,,,false,2022-11-01 09:37:06.009000,2022-11-01 09:37:06.009000,,,,,
47c06a6c-1b23-4d5d-a20b-308123e1c930,57ebeba5-d94c-425d-9e3d-b4e1f675119e,b056af51-fc49-4c76-b536-a041d5766d34,ID_CARD,FRONT,da24994b-799e-43a8-9227-a631555b14f1,mock,ACCEPTED,id_card_mock_front.png,uploaded-9b8630a2-a827-4c27-905b-...,1a5ea207-1608-4d83-aec7-c506317253b0,extracted-photo-id,,,,,true,2022-11-01 09:36:27.691000,2022-11-01 09:36:27.691000,2022-11-01 09:36:35.007000,,2022-11-01 09:36:35.007000,,
da24994b-799e-43a8-9227-a631555b14f1,57ebeba5-d94c-425d-9e3d-b4e1f675119e,b056af51-fc49-4c76-b536-a041d5766d34,ID_CARD,BACK,47c06a6c-1b23-4d5d-a20b-308123e1c930,mock,ACCEPTED,id_card_mock_back.png,uploaded-4986d63b-6cd7-4dcd-afcf-...,1a5ea207-1608-4d83-aec7-c506317253b0,extracted-photo-id,,,,,true,2022-11-01 09:36:27.691000,2022-11-01 09:36:27.691000,2022-11-01 09:36:35.007000,,2022-11-01 09:36:35.007000,,
```

# Solution

- [x] set `isDocumentVerificationOnSubmitEnabled=false`
- [ ] cross validate both documents via `api/investigate/samples` and send this investigation ID to client evaluation
- [x] remove query parameter `uploadSessionId`

### PR DESCRIPTION
**Pull request by banterCZ**

### CODE REVIEW

File: enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/service/AuditService.java [hunk 1]
```diff
@@ -250,15 +251,16 @@ private static AuditDetail createAuditDetail(final AuditType type, final Identit
...
+        final IdentityVerificationEntity identityVerification = document.getIdentityVerification();
         return AuditDetail.builder()
                 .type(AuditType.DOCUMENT_VERIFICATION.code)
                 .param(IDENTITY_VERIFICATION_ID, identityVerification.getId())
                 .param(PROCESS_ID, identityVerification.getProcessId())
                 .param(ACTIVATION_ID, identityVerification.getActivationId())
                 .param(USER_ID, identityVerification.getUserId())
-                .param(DOCUMENT_VERIFICATION_ID, documentVerification.getId())
+                .param(DOCUMENT_ID, document.getId())
+                .param(DOCUMENT_VERIFICATION_ID, document.getVerificationId())
```
Comment (banterCZ) [inline enrollment-server-onboarding-common/src/main/java/com/wultra/app/onboardingserver/common/service/AuditService.java:263]: 👀 To audit internal and external ID!

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/docverify/zenid/config/ZenidConfigProps.java [hunk 2]
```diff
@@ -87,4 +89,10 @@ public class ZenidConfigProps {
     public Optional<String> getProfile() {
         return Optional.ofNullable(profile);
     }
+
+    @PostConstruct
+    void validate() {
+        // TODO (racansky, 2022-11-03) DocumentVerificationFinalAction works only in sync mode
+        Assert.state(!asyncProcessingEnabled, "ZenId async functionality is not fully supported yet");
```
Comment (banterCZ) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/docverify/zenid/config/ZenidConfigProps.java:96]: As we agreed, add safety check not to allow configuring async processing.

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentVerificationService.java [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+        }
+    }
+
+    private void accept(
+            final IdentityVerificationEntity identityVerification,
+            final List<DocumentVerificationEntity> documentVerifications,
+            final OwnerId ownerId) {
+        documentVerifications.forEach(docVerification ->
+            auditService.audit(docVerification, "Document accepted at final verification for user: {}", identityVerification.getUserId()));
+        identityVerificationService.moveToPhaseAndStatus(identityVerification, DOCUMENT_VERIFICATION_FINAL, ACCEPTED, ownerId);
```
Comment (banterCZ) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentVerificationService.java:138]: Not introduced as a separate state of state machine configuration yet. Added just for auditing reasons. For sync processing, this step makes sense to be done all at once. Should be added to the diagram.

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentVerificationService.java [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+        identityVerificationService.moveToPhaseAndStatus(identityVerification, DOCUMENT_VERIFICATION_FINAL, IN_PROGRESS, ownerId);
+
+        final List<DocumentVerificationEntity> documentVerifications = filterDocumentVerifications(identityVerification);
+
+        final List<String> uploadIds = documentVerifications.stream()
+                .map(DocumentVerificationEntity::getUploadId)
+                .collect(toList());
+
+        final DocumentsVerificationResult result = documentVerificationProvider.verifyDocuments(ownerId, uploadIds);
```
Comment (romanstrobl) [inline enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentVerificationService.java:104]: In case of an error during `verifyDocuments` call the identity verification may get stuck in this phase/status. Let's raise a ticket for handling this separately.

**Review by romanstrobl — APPROVED** [review]
This pull request should work as expected for common scenarios. There are some corner cases, I raised issues for them and left one for review. Let's proceed with testing and handle any additional changes based on test results as another pull request.

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentVerificationService.java (hunk 4 above)
Comment (romanstrobl) [reply enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentVerificationService.java:104]: Raised as: https://github.com/wultra/enrollment-server/issues/496

File: enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentVerificationService.java [hunk 5]
```diff
@@ -0,0 +1,201 @@
...
+        identityVerification.setTimestampFailed(ownerId.getTimestamp());
+
+        identityVerificationService.moveToPhaseAndStatus(identityVerification, identityVerification.getPhase(), FAILED, ownerId);
+
+        incrementErrorScore(identityVerification, OnboardingProcessError.ERROR_DOCUMENT_VERIFICATION_FAILED, ownerId);
+    }
+
+    private void incrementErrorScore(
+            final IdentityVerificationEntity identityVerification,
+            final OnboardingProcessError errorDocumentVerificationFailed,
```
[the remark this answers is not in the document]
Comment (banterCZ) [reply enrollment-server-onboarding/src/main/java/com/wultra/app/onboardingserver/impl/service/document/DocumentVerificationService.java:187]: 👀 Good catch, too fast method extraction.
